UnitsNet/QuantityValue.cs

```diff
@@ -90,10 +113,235 @@ private QuantityValue(decimal val)
+            }
+            else if (b.IsDecimal)
+            {
+                return (decimal)a._value.GetValueOrDefault(0) < b._valueDecimal.GetValueOrDefault();
+            }
+            else
+            {
+                return a._value < b._value;
+            }
+        }
+
+        /// <summary>
+        /// Greater-than-or-equal operator
+        /// </summary>
+        public static bool operator >=(QuantityValue a, QuantityValue b)
+        {
+            if (a.IsDecimal && b.IsDecimal)
+            {
+                return a._valueDecimal.GetValueOrDefault() >= b._valueDecimal.GetValueOrDefault();
+            }
+            else if (a.IsDecimal)
+            {
+                return a._valueDecimal.GetValueOrDefault() >= (decimal)b._value.GetValueOrDefault(0); // other._value cannot be null here
+            }
+            else if (b.IsDecimal)
+            {
+                return (decimal)a._value.GetValueOrDefault(0) >= b._valueDecimal.GetValueOrDefault();
+            }
+            else
+            {
+                return a._value >= b._value;
+            }
+        }
+
+        /// <summary>
+        /// Less-than-or-equal operator
+        /// </summary>
+        public static bool operator <=(QuantityValue a, QuantityValue b)
+        {
+            if (a.IsDecimal && b.IsDecimal)
+            {
+                return a._valueDecimal.GetValueOrDefault() <= b._valueDecimal.GetValueOrDefault();
+            }
+            else if (a.IsDecimal)
+            {
+                return a._valueDecimal.GetValueOrDefault() <= (decimal)b._value.GetValueOrDefault(0); // other._value cannot be null here
+            }
+            else if (b.IsDecimal)
+            {
+                return (decimal)a._value.GetValueOrDefault(0) <= b._valueDecimal.GetValueOrDefault();
+            }
+            else
+            {
+                return a._value <= b._value;
+            }
+        }
+
+        /// <summary>
+        /// Returns the negated value of the operand
+        /// </summary>
+        /// <param name="v">Value to negate</param>
+        /// <returns>-v</returns>
+        public static QuantityValue operator -(QuantityValue v)
+        {
+            if (v._valueDecimal.HasValue)
+            {
+                return new QuantityValue(-v._valueDecimal.Value);
+            }
+            else
+            {
+                return new QuantityValue(-v._value.GetValueOrDefault());
+            }
+        }
+
+        /// <summary>
+        /// Multiplication operator. Note that this performs an operation on a raw value, as no unit is associate the the value at this point.
+        /// </summary>
+        public static QuantityValue operator *(QuantityValue a, QuantityValue b)
+        {
+            if (a._valueDecimal.HasValue && b._valueDecimal.HasValue)
+            {
+                return new QuantityValue(a._valueDecimal.Value * b._valueDecimal.Value);
+            }
+            else
```

Comment:
Don't we have to check the two other `else if` cases too, to avoid calculating with unassigned values? Same below.

Reply:
True, I overlooked that.